lib/Backends/Interpreter/InterpreterDeviceManager.h

```diff
@@ -17,14 +17,23 @@
 #define GLOW_BACKENDS_INTERPRETER_INTERPRETERDEVICEMANAGER_H
 
 #include "glow/Backends/QueueBackedDeviceManager.h"
+#include <future>
 
 namespace glow {
+
+using TensorPipeTy = std::pair<std::future<Tensor>, std::promise<Tensor>>;
+using TensorPipeMapTy = std::map<uintptr_t, TensorPipeTy>;
```

Comment:
Because lower_bound/upper_bound algorithm are not required then using 
using TensorPipeMapTy = std::unordered_map<uintptr_t, TensorPipeTy>;
would be more efficient.